optashift-employee-rostering-webapp/pom.xml

```diff
@@ -28,135 +30,159 @@
+      <artifactId>optashift-employee-rostering-server</artifactId>
     </dependency>
     <dependency>
       <groupId>org.optaplanner.openshift</groupId>
-      <artifactId>optashift-employee-rostering-server</artifactId>
+      <artifactId>optashift-employee-rostering-gwtui</artifactId>
+      <type>war</type>
+      <scope>runtime</scope>
     </dependency>
     <dependency>
-      <groupId>javax</groupId>
-      <artifactId>javaee-api</artifactId>
-      <scope>provided</scope>
+      <groupId>junit</groupId>
+      <artifactId>junit</artifactId>
+      <scope>test</scope>
     </dependency>
-    <!-- Logging -->
     <dependency>
       <groupId>org.slf4j</groupId>
       <artifactId>slf4j-api</artifactId>
     </dependency>
     <dependency>
-      <groupId>ch.qos.logback</groupId>
-      <artifactId>logback-classic</artifactId>
-      <scope>runtime</scope>
+      <groupId>org.slf4j</groupId>
+      <artifactId>slf4j-simple</artifactId>
+    </dependency>    
+    <dependency>
+      <groupId>org.jboss.arquillian.junit</groupId>
+      <artifactId>arquillian-junit-core</artifactId>
+      <scope>test</scope>
     </dependency>
-    
-     <!-- Testing -->
     <dependency>
-      <groupId>junit</groupId>
-      <artifactId>junit</artifactId>
+      <groupId>org.jboss.arquillian.core</groupId>
+      <artifactId>arquillian-core-impl-base</artifactId>
       <scope>test</scope>
     </dependency>
-
     <dependency>
-      <groupId>org.mockito</groupId>
-      <artifactId>mockito-core</artifactId>
+      <groupId>org.jboss.arquillian.test</groupId>
+      <artifactId>arquillian-test-impl-base</artifactId>
       <scope>test</scope>
     </dependency>
-
     <dependency>
-      <groupId>org.apache.httpcomponents</groupId>
-      <artifactId>httpclient</artifactId>
+      <groupId>org.jboss.arquillian.container</groupId>
+      <artifactId>arquillian-container-impl-base</artifactId>
       <scope>test</scope>
     </dependency>
-
     <dependency>
-      <groupId>org.jboss.arquillian.junit</groupId>
-      <artifactId>arquillian-junit-container</artifactId>
+      <groupId>org.jboss.arquillian.core</groupId>
+      <artifactId>arquillian-core-api</artifactId>
       <scope>test</scope>
     </dependency>
-
     <dependency>
       <groupId>org.wildfly.arquillian</groupId>
       <artifactId>wildfly-arquillian-container-managed</artifactId>
       <scope>test</scope>
     </dependency>
 
     <dependency>
+      <groupId>org.jboss.remoting</groupId>
+      <artifactId>jboss-remoting</artifactId>
+      <version>${arquillian.version.org.jboss.remoting}</version>
+      <scope>test</scope>
+    </dependency>
+    
+   <dependency>
       <groupId>org.jboss.shrinkwrap.resolver</groupId>
       <artifactId>shrinkwrap-resolver-depchain</artifactId>
       <type>pom</type>
       <scope>test</scope>
     </dependency>
-
-
+    
     <dependency>
       <groupId>commons-logging</groupId>
       <artifactId>commons-logging</artifactId>
```

Comment:
We can't have multiple logging implementations in our classpath.
There can only be slf4j-api (= the api) in all modules and logback (the implementation) only in end-user modules (the war and the benchmarker). All other logging jars, such as slf4j-simple and commons-logging are evil.
slf4j bridges are also allowed (they make sure that if we depend on spring (we don't) and spring logs to commons-logging (it doesn't) that spring's log messages get bridged into slf4j-api and so respect our logback.xml configuration.